tools/xabuild/XABuildPaths.cs

```diff
@@ -156,12 +154,13 @@ public XABuildPaths ()
 				}
 				NuGetRestoreTargets = Path.Combine (MSBuildBin, "NuGet.targets");
 				if (!File.Exists (NuGetRestoreTargets) && !string.IsNullOrEmpty (DotNetSdkPath)) {
-					NuGetRestoreTargets  = Path.Combine (DotNetSdkPath, "..", "NuGet.targets");
+					NuGetRestoreTargets = Path.Combine (DotNetSdkPath, "..", "NuGet.targets");
 				}
 			}
 
 			FrameworksDirectory       = Path.Combine (prefix, "xbuild-frameworks");
 			MSBuildExtensionsPath     = Path.Combine (prefix, "xbuild");
+			SystemTargetsDirectories  = new [] { Path.Combine (MSBuildPath, vsVersion), Path.Combine (MSBuildPath, "Microsoft") };
```

Comment:
um you will have `$mono_prefix/lib/mono/msbuild/15.0` and `$mono_prefix/lib/mono/msbuild/Microsoft` here, but the correct path has `xbuild` like `$mono_prefix/lib/mono/xbuild/15.0`.